Doc/library/itertools.rst

```diff
@@ -1345,6 +1346,16 @@ The following recipes have a more mathematical flavor:
     Traceback (most recent call last):
     ...
     ValueError
+    >>> # Verify that both paths can find identical NaN values
+    >>> x = float('NaN')
+    >>> y = float('NaN')
+    >>> list(iter_index([0, x, x, y, 0], x))
+    [1, 2]
+    >>> list(iter_index(iter([0, x, x, y, 0]), x))
+    [1, 2]
+    >>> # Test list input. Lists do not support None for the stop argument
+    >>> list(iter_index(list('AABCADEAF'), 'A'))
+    [0, 1, 4, 7]
```

Comment:
Was not list input already tested above, in `list(iter_index([0, x, x, y, 0], x))`?

If you want a separate test for list which does not involve NaN, it is better to move this test upper, just after `list(iter_index(iter('AABCADEAF'), 'A', 1, 7))`.